framework/reward_net.py

```diff
@@ -27,21 +27,19 @@ def preference_prob(self, r1, r2):
         exp1 = torch.exp(torch.sum(r1))
         exp2 = torch.exp(torch.sum(r2))
         prop1 = exp1 / (exp1 + exp2)
-        prop2 = exp2 / (exp1 + exp2)
-        return prop1, prop2
+        return prop1
 
     def preference_loss(self, predictions, preferences):
-        # Compute loss based on human feedback
-        # predictions shape: (batch_size, 2)
-        # preferences shape: (batch_size, 2)
-        return -torch.sum(preferences * torch.log(predictions))
+        # Compute binary cross entropy loss based on human feedback
+        return -torch.mean(preferences * torch.log(predictions) + (1 - preferences) * torch.log(1 - predictions))
```

Comment:
Why do we need the mean here?

Reply:
Currently, we process one preference-prediction pair at a time, so `torch.mean` is redundant. If we later extend this process to handle batches, `torch.mean` will compute the average loss across the batch.